src/views/TabView/TabBarBottom.js

```diff
@@ -158,6 +165,68 @@ class TabBarBottom extends React.PureComponent<Props> {
     return testIDProps;
   };
 
+  _tabItemMaxWidth() {
+    const { tabStyle, layout } = this.props;
+    let maxTabBarItemWidth;
+
+    const flattenedTabStyle = StyleSheet.flatten(tabStyle);
+
+    if (flattenedTabStyle) {
+      if (typeof flattenedTabStyle.width === 'number') {
+        maxTabBarItemWidth = flattenedTabStyle.width;
+      } else if (
+        typeof flattenedTabStyle.width === 'string' &&
+        flattenedTabStyle.endsWith('%')
+      ) {
+        const width = parseFloat(flattenedTabStyle.width);
+        if (Number.isFinite(width)) {
+          maxTabBarItemWidth = layout.width * (width / 100);
+        }
+      } else if (typeof flattenedTabStyle.maxWidth === 'number') {
+        maxTabBarItemWidth = flattenedTabStyle.maxWidth;
+      } else if (
+        typeof flattenedTabStyle.maxWidth === 'string' &&
+        flattenedTabStyle.endsWith('%')
+      ) {
+        const width = parseFloat(flattenedTabStyle.maxWidth);
+        if (Number.isFinite(width)) {
+          maxTabBarItemWidth = layout.width * (width / 100);
+        }
+      }
+    }
+
+    if (!maxTabBarItemWidth) {
+      maxTabBarItemWidth = defaultMaxTabBarItemWidth;
+    }
+
+    return maxTabBarItemWidth;
+  }
+
+  _shouldUseHorizontalTabs() {
+    const { routes } = this.props.navigation.state;
+    const { isLandscape, layout, adaptive, tabStyle } = this.props;
+
+    if (!adaptive) {
+      return false;
+    }
+
+    let tabBarWidth = layout.width;
+    if (tabBarWidth === 0) {
+      return isTablet;
+    }
+
+    const isWidthConstrained = layout.width < 450;
```

Comment:
can we just use horizontal tabs 1) on ipad always 2) on iphone when in landscape?

no need for magic numbers like 450 here

Reply:
No, because on tablets in split screen mode, the app could be as wide as it would be on a phone, and there is no space for horizontal tabs (this was the original issue #2794).  We need to be able to handle all sorts of split-screen configurations, Platform.isPad is useless for this because it can't detect split-screen.  I believe split-screen on Android is becoming more popular as well.

Reply:
Please check original conversion in #3041 for more details on motivation for change and methodology chosen.